### DISCUSSION TITLE
App tweak to a near perfect OS

### DISCUSSION THREAD

**Discussion by kiltiking** [linuxmint/discussions#374] [Ideas]
When I open System Settings and increase its size, I then open one of the settings options, when I hit the back arrow to return to the main window, it reverts back to the default size. I have a 27 inch monitor and Control Center looks like a postage stamp which I have to re-size every time!  It's not a big deal but if all apps remembered their size and position it would  be a nicer user experience.

**Comment by claudiux** [comment]
There is a hack (but it isn't entirely satisfactory).

Edit with the root rights the `/usr/share/cinnamon/cinnamon-settings/cinnamon-settings.py` script then change values in lines 45 and 46.

```
WIN_WIDTH = 1000 # instead of 800
WIN_HEIGHT = 800 # instead of 600
```
(Set with your own values.)

Also line 763:
`self.window.resize(WIN_WIDTH, WIN_HEIGHT)` instead of `self.window.resize(800, 600)`

Save this file and restart Cinnamon. That works.

**Reply by kiltiking** [reply]
Thank you...while it still starts small, once I click on one of the setting options it adopts the custom size and stays that way. Thanks again.

**Reply by 100savage** [reply]
Thank you for sharing this. I always thought it opened a little small. I changed lines 45/46 and line 259. 
Line 259 fixed the window on first opening.
Lines 45/46 fixed the selected app inside system settings.

**Reply by claudiux** [reply]
Yes, line 259 too. I'd forgotten about this change.

**Reply by kiltiking** [reply]
Perfect...thanks folks...I'm glad I joined Github. One more question...does every app (e.g. Software Manager) have a .py file that can be edited ? I'd have to try get into python if so...

**Reply by claudiux** [reply]
Take a look at https://github.com/linuxmint/mintinstall. This will tell you where the files are on your system. Be careful when you modify them!

**Reply by kiltiking** [reply]
This is great...I got the source files, had a look at the directories etc. and then modified Software Manager to open larger by changing lines 80 & 81 in the file /usr/share/linuxmint/mintinstall/mintinstall.glade. Over the weekend I'm going to set up a separate PC to learn Python and dabble with apps and the OS. Thanks again.

**Reply by claudiux** [reply]
@kiltiking 
Back up your modified files on other media. During system updates, your changes may be lost. For an excellent file comparison tool, install the `nemo-compare` package.
